### PR TITLE
[Elastic-SAN] 2023-01-01 api-version public preview

### PR DESCRIPTION
**Pull request by calvinhzy**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Elastic-SAN

File: src/elastic-san/azext_elastic_san/custom.py [hunk 1]
```diff
@@ -9,6 +9,72 @@
...
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg, AAZResourceIdArgFormat
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.identity.user_assigned_identity_id = AAZStrArg(
+            options=["user-assigned-identity"],
+            help="The ARM resource identifier of the User Assigned identity that will be used with this volume group. ",
+            fmt=AAZResourceIdArgFormat(template="/subscriptions/{subscription}/resourceGroups/{resource_group}"
+                                                "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{}")
+        )
```
Comment (necusjz) [inline src/elastic-san/azext_elastic_san/custom.py:54]: Is it nullable?

File: src/elastic-san/azext_elastic_san/custom.py [hunk 2]
```diff
@@ -9,6 +9,72 @@
...
+        args_schema.identity.user_assigned_identities._registered = False
+        args_schema.encryption_properties.identity.user_assigned_identity._fmt = \
+            AAZResourceIdArgFormat(template="/subscriptions/{subscription}/resourceGroups/{resource_group}"
+                                            "/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{}")
+        return args_schema
+
+    def pre_operations(self):
+        args = self.ctx.args
+        if args.identity.user_assigned_identity_id:
```
Comment (necusjz) [inline src/elastic-san/azext_elastic_san/custom.py:37]: Why not use `has_value`?

File: src/elastic-san/azext_elastic_san/custom.py [hunk 3]
```diff
@@ -9,6 +9,72 @@
...
+        if instance.identity and instance.identity.user_assigned_identities and \
+                args.identity and args.identity.user_assigned_identities:
+            uai_id_new = list(args.identity.user_assigned_identities.keys())[0]
+            uai_id_old = list(instance.identity.user_assigned_identities.keys())[0]
+            if uai_id_old != uai_id_new:
+                if instance.properties and instance.properties.encryption_properties and \
+                        instance.properties.encryption_properties.key_vault_properties and \
+                        args.encryption_properties and not args.encryption_properties.key_vault_properties:
+                    args.encryption_properties.key_vault_properties = \
+                        instance.properties.encryption_properties.key_vault_properties
```
Comment (necusjz) [inline src/elastic-san/azext_elastic_san/custom.py:80]: Same concern about `has_value`.

File: src/elastic-san/azext_elastic_san/custom.py (hunk 1 above)
Comment (calvinhzy) [reply src/elastic-san/azext_elastic_san/custom.py:55]: Not according to service right now, might add in future releases.

File: src/elastic-san/azext_elastic_san/custom.py (hunk 3 above)
Comment (calvinhzy) [reply src/elastic-san/azext_elastic_san/custom.py]: Fixed

File: src/elastic-san/azext_elastic_san/custom.py (hunk 2 above)
Comment (calvinhzy) [reply src/elastic-san/azext_elastic_san/custom.py]: Fixed

**Review by necusjz — APPROVED** [review]

**Pull request merged**